### PR TITLE
feat: add C implementation for `math/base/special/bernoulli`

### LINKED ISSUE #1664: [RFC]: Add C implementation for `@stdlib/math/base/special/bernoulli`
### Description

This RFC proposes adding native C implementation for `@stdlib/math/base/special/bernoulli`
```
double stdlib_base_bernoulli( const int n );
```

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by gunjjoshi**
Resolves #1664 

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for [`math/base/special/bernoulli`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/bernoulli).

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves a part of #649 
-   fixes #1664 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   ☑️ Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by gunjjoshi** [comment]
Since there was inactivity on #1664 due to a dependency's C implementation not being done, and all its dependencies' C installation are done now, so I've made this PR.

File: lib/node_modules/@stdlib/math/base/special/bernoulli/benchmark/benchmark.native.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+// MODULES //
+
+var resolve = require( 'path' ).resolve;
+var bench = require( '@stdlib/bench' );
+var randu = require( '@stdlib/random/base/randu' );
+var floor = require( '@stdlib/math/base/special/floor' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var tryRequire = require( '@stdlib/utils/try-require' );
+var pkg = require( '@stdlib/math/base/special/bernoulli/package.json' ).name;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/bernoulli/benchmark/benchmark.native.js]:
```suggestion
var pkg = require( './../package.json' ).name;
```

File: lib/node_modules/@stdlib/math/base/special/bernoulli/lib/native.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+* // returns ~-529.124
+*
+* @example
+* var y = bernoulli( 260 );
+* // returns -Infinity
+*
+* @example
+* var y = bernoulli( 262 );
+* // returns Infinity
+*
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/bernoulli/lib/native.js]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/bernoulli/src/main.c [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  -2.22448916821798346676602348865048510824835e277,
+  3.28626791906901391668189736436895275365183e280,
+  -4.93559289559603449020711938191575963496999e283,
+  7.53495712008325067212266049779283956727824e286,
+  -1.16914851545841777278088924731655041783900e290,
+  1.84352614678389394126646201597702232396492e293,
+  -2.95368261729680829728014917350525183485207e296,
+  4.80793212775015697668878704043264072227967e299,
+  -7.95021250458852528538243631671158693036798e302,
+  1.33527841873546338750122832017820518292039e306
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/bernoulli/src/main.c]: TAB indentation

File: lib/node_modules/@stdlib/math/base/special/bernoulli/src/main.c [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+* @example
+* double out = stdlib_base_bernoulli( 0 );
+* // returns 1
+*/
+double stdlib_base_bernoulli( const int32_t n ) {
+	if ( stdlib_base_is_nan( n ) || !stdlib_base_is_nonnegative_integer( n ) ) {
+		return 0.0 / 0.0; // NaN
+	}
+	if ( stdlib_base_is_odd( n ) ) {
+		return 0;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/bernoulli/src/main.c]:
```suggestion
		return 0.0;
```

File: lib/node_modules/@stdlib/math/base/special/bernoulli/src/main.c [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+  7.53495712008325067212266049779283956727824e286,
+  -1.16914851545841777278088924731655041783900e290,
+  1.84352614678389394126646201597702232396492e293,
+  -2.95368261729680829728014917350525183485207e296,
+  4.80793212775015697668878704043264072227967e299,
+  -7.95021250458852528538243631671158693036798e302,
+  1.33527841873546338750122832017820518292039e306
+};
+
+// VARIABLES //
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/bernoulli/src/main.c]:
```suggestion
```

We do not write these in C files

File: lib/node_modules/@stdlib/math/base/special/bernoulli/src/main.c [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+
+int32_t MAX_BERNOULLI = 258;
+
+// MAIN //
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/bernoulli/src/main.c]: same here

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Minor changes requested, once done this will be ready to merge.

**Comment by gunjjoshi** [comment]
I have made the changes as per your suggestions. Thanks for pointing them out @Pranavchiku  !

File: lib/node_modules/@stdlib/math/base/special/bernoulli/src/main.c [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+	0.166666666666666666666666666666666666666667,
+	-0.0333333333333333333333333333333333333333333,
+  	0.0238095238095238095238095238095238095238095,
+  	-0.0333333333333333333333333333333333333333333,
+  	0.0757575757575757575757575757575757575757576,
+  	-0.253113553113553113553113553113553113553114,
+  	1.16666666666666666666666666666666666666667,
+  	-7.09215686274509803921568627450980392156863,
+  	54.9711779448621553884711779448621553884712,
+  	-529.124242424242424242424242424242424242424,
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/bernoulli/src/main.c]:
```suggestion
	0.0238095238095238095238095238095238095238095,
	-0.0333333333333333333333333333333333333333333,
	0.0757575757575757575757575757575757575757576,
	-0.253113553113553113553113553113553113553114,
	1.16666666666666666666666666666666666666667,
	-7.09215686274509803921568627450980392156863,
	54.9711779448621553884711779448621553884712,
	-529.124242424242424242424242424242424242424,
```

**Review by Pranavchiku — APPROVED** [review]
Works well locally. Looks good to me. Thanks @gunjjoshi !

**Pull request merged**